spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/cassandra/CassandraVectorStoreAutoConfiguration.java

```diff
@@ -36,7 +36,7 @@
  * @since 1.0.0
  */
 @AutoConfiguration(after = CassandraAutoConfiguration.class)
-@ConditionalOnClass({ CassandraVectorStore.class, EmbeddingClient.class, CqlSession.class })
+@ConditionalOnClass({ CassandraVectorStore.class, CqlSession.class })
```

Comment:
If it is not required then `EmbeddingClient embeddingClient` should be wrapped with `ObjectProvider`

```java
public CassandraVectorStore vectorStore(ObjectProvider<EmbeddingClient> embeddingClient, CassandraVectorStoreProperties properties,
			CqlSession cqlSession) {
```

otherwise, it will be a required bean at it will fail if not present

Reply:
that was what I first thought.

but when you specify an EmbeddingClient (in a constructor) so that that bean is instantiated, then the CassandraVectorStore bean is automatically instantiated too.

that isn't desired.  a bean that's not needed shouldn't be instantiated.

this code works with this patch.  note that i don't declare or configure the EmbeddingClient bean anywhere…
https://github.com/datastax/ai-agent-java/blob/main/src/main/java/com/datastax/ai/agent/AiApplication.java 

I also noted that other StoreAutoConfigurations don't have it:
e.g. https://github.com/spring-projects/spring-ai/blob/main/spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/pgvector/PgVectorStoreAutoConfiguration.java 
(also HanaDb)

Reply:
Please ignore my first comment, you don't need to do that unless the bean is not available. However, in the project the bean is created because [spring-ai-openai-spring-boot-starter is present](https://github.com/datastax/ai-agent-java/blob/main/pom.xml#L50-L54). I wonder if `@ConditionalOnBean(EmbeddingClient.class)` is needed.